docs/general/networking/fail2ban.md

```diff
@@ -86,3 +86,197 @@ Assuming you've at least one failed authentication attempt, you can test this ne
+
+---
+
+### Step One: Create the Jail
+
+You need to create a jail for Fail2Ban. If you're on Ubuntu and use nano as an editor, run:
+
+```bash
+sudoedit /etc/fail2ban/jail.d/jellyfin.local
+```
+
+Add this to the new file, replacing `/path_to_logs` with the path to the log files above, e.g., `/var/log/jellyfin/`:
+
+```ini
+[jellyfin]
+
+backend = auto
+enabled = true
+port = 80,443
+protocol = tcp
+filter = jellyfin
+maxretry = 3
+bantime = 86400
+findtime = 43200
+logpath = /path_to_logs/jellyfin*.log
+```
+
+Save and exit.
+
+**Note:**  
+If Jellyfin is running in a Docker container, add the following line to the `jellyfin.local` file to ensure compatibility with Docker's networking:
+
+```bash
+action = iptables-allports[name=jellyfin, chain=DOCKER-USER]
+```
+
+### Step Two: Create the Filter
+
+The filter contains a set of rules that Fail2Ban will use to identify failed authentication attempts. Create the filter by running:
+
+```bash
+sudoedit /etc/fail2ban/filter.d/jellyfin.conf
+```
+
+Add:
+
+```ini
+[Definition]
+failregex = ^.*Authentication request for .* has been denied \(IP: "<ADDR>"\)\.
+```
+
+Save and restart Fail2Ban:
+
+```bash
+sudo systemctl restart fail2ban
+```
+
+Verify Fail2Ban is running:
+
+```bash
+sudo systemctl status fail2ban
+```
+
+### Step Three: Test
+
+To test this setup with existing logs:
+
+```bash
+sudo fail2ban-regex /path_to_logs/*.log /etc/fail2ban/filter.d/jellyfin.conf --print-all-matched
+```
+
+---
+
+## Advanced Fail2Ban Setup: Managing Bans on an Upstream Proxy Server
+
+To enhance security, Fail2Ban can manage IP bans on an upstream reverse proxy server instead of directly on the Jellyfin server. This setup allows you to block malicious IPs closer to your network’s entry point, potentially benefiting other services using the same proxy.
+
+This guide offers two configurations for setting up Fail2Ban to manage IP bans on an upstream reverse proxy server:
+
+1. **Dynamic Chains**: Each Fail2Ban jail creates and manages its own `iptables` chain on the upstream server.
+2. **Communal Chain**: All Fail2Ban jails share a single `iptables` chain on the upstream server.
+
+### Prerequisites
+
+- **Fail2Ban** is installed on your local server (where services such as Jellyfin, Nginx, etc., are running).
+- **SSH key-based access** is set up from your Fail2Ban server to the upstream server.
+- **iptables** is configured on the upstream server.
+
+### Step 1: Set Up SSH Key-Based Authentication
+
+Establish SSH access from the Fail2Ban server to the upstream server for automated IP ban/unban actions.
+
+1. **Generate an SSH Key** (if not already created):
+
+   ```bash
+   ssh-keygen -t rsa -b 4096 -f /root/.ssh/id_rsa
+   ```
+
+2. **Copy the Key to the Upstream Server**:
+
+   ```bash
+   ssh-copy-id -i /root/.ssh/id_rsa.pub root@<upstream-server-ip>
+   ```
+
+3. **Verify Access**:
+
+   ```bash
+   ssh -i /root/.ssh/id_rsa root@<upstream-server-ip>
+   ```
+
+---
```

Comment:
why is this both a prerequisite and part of the guide? If you would like to include this here, please remove this from the prerequisites

Reply:
good point - will remove from prereqs.